dali/base/dadfs.cpp

```diff
@@ -5882,6 +5904,34 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
             throw MakeStringException(-1,"addSubFile: File %s is already a subfile of %s", sub->queryLogicalName(),queryLogicalName());
     }
 
+    void validate()
+    {
+        unsigned numSubfiles = root->getPropInt("@numsubfiles",0);
+        if (numSubfiles)
+        {
+            Owned<IPropertyTreeIterator> treeIter = root->getElements("SubFile");
+            unsigned subFileCount = 0;
+            ForEach(*treeIter)
+            {
+                IPropertyTree & st = treeIter->query();
+                StringBuffer subfilename;
+                st.getProp("@name", subfilename);
+                if (!parent->exists(subfilename.str(), NULL))
+                    throw MakeStringException(-1, "Logical subfile '%s' doesn't exists!", subfilename.str());
+
+                if (!parent->isSuperFile(subfilename.str(), NULL, NULL, 0))
```

Comment:
trivial: you could argue that adding the default params in the implementation of isSuperFile() to mirror the interface method definition would be better than adding extra ,NULL, 0 here.